src/PAModelTests/DataSourceTests.cs

```diff
@@ -97,7 +97,38 @@ public void TestNoLocalDatabaseRefsWhenLocalDatabaseReferencesPropertyWasEmptyJs
             Assert.IsFalse(loadedMsApp._entropy.LocalDatabaseReferencesAsEmpty);
             Assert.IsTrue(loadedMsApp._dataSourceReferences.Count == 0);
         }
-       
+
+        [DataTestMethod]
+        [DataRow("EmptyLocalDBRefsHashMismatchProperties.msapp")]
+        public void TestConnectionInstanceIDHandling(string appName)
+        {
+            var pathToMsApp = Path.Combine(Environment.CurrentDirectory, "Apps", appName);
+            Assert.IsTrue(File.Exists(pathToMsApp));
+            
+            var (msApp, errors) = CanvasDocument.LoadFromMsapp(pathToMsApp);
+            errors.ThrowOnErrors();
+
+            using var sourcesTempDir = new TempDir();
+            var sourcesTempDirPath = sourcesTempDir.Dir;
+            msApp.SaveToSources(sourcesTempDirPath);
+
+            var loadedMsApp = SourceSerializer.LoadFromSource(sourcesTempDirPath, new ErrorContainer());
+
+            // Testing if conn instance id is added to entropy
+            Assert.IsNotNull(loadedMsApp._entropy.LocalConnectionIDReferences);
+
+            using (var tempFile = new TempFile())
+            {
+                // Repack the app
+                MsAppSerializer.SaveAsMsApp(loadedMsApp, tempFile.FullPath, new ErrorContainer());
+
+                // Comparing original .msapp with repacked .msapp
+                // and testing no deltas 
+                bool ok = MsAppTest.MergeStressTest(pathToMsApp, tempFile.FullPath);
```

Comment:
Talked offline - this is a pretty heavy operation (makes the unit test take 4ish minutes) and I think instead we can evaluate the `ErrorContainer` that comes back from `SaveToSources` on line 113.